tests/test_i2c.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+# Copyright 2016-2024 XMOS LIMITED.
+# This Software is subject to the terms of the XMOS Public Licence: Version 1.
+import pytest
+import utils
+import subprocess
+from pathlib import Path
+
+def test_i2c_device():
+    """
+    This test runs on the device using xsim.
+    It creates several I2C control interfaces, it sends some read and write commands, and it checks that the responses are correct.
+    """
+    target = "i2c_device"
+    xe_path = utils.build_firmware(target, project_dir=Path(__file__).parent / target, build_dir="build")
+    output = None
+    try:
+        output = utils.xsim_firmware(xe_path)
```

Comment:
same comment I would prefer intead of try except bloc, assert(p.returncode == 0)

Reply:
Fixed by multiple commits.